x-pack/plugins/alerting/server/rules_client/methods/clone.ts

```diff
@@ -117,6 +117,7 @@ export async function clone<Params extends RuleTypeParams = never>(
     mutedInstanceIds: [],
     executionStatus: getRuleExecutionStatusPending(lastRunTimestamp.toISOString()),
     monitoring: getDefaultMonitoring(lastRunTimestamp.toISOString()),
+    revision: 0, // TODO: Clarify if we're resetting revision since it's a new rule, or carrying over from previous rule (existing security solution behavior)
```

Comment:
Open question: should a rule clone/duplicate reset `revision` to `0` or keep the current value? Current Security behavior has been to keep the current value, but we should align here for consistency (no strong opinion on the security side).

Reply:
When creating new rules, it's logical to set the revision number to 0 because revisions are used to track the history of changes, and a new rule doesn't have any previous history. Moreover, suppose we were to display the revision history to the user through a dropdown menu, allowing them to select which historical revision to revert to. In that case, it could be counterintuitive if the revision numbers started from a value other than zero, which might look like a bug. Like there are some historical versions I don't have access to.

Reply:
While testing locally, I noticed duplicated rules (through bulk actions) have revision set to 1. So it seems like their revision gets immediately bumped after creation. When I create rules one by one they have revision set to 0.

Reply:
> Moreover, suppose we were to display the revision history to the user through a dropdown menu, allowing them to select which historical revision to revert to. In that case, it could be counterintuitive if the revision numbers started from a value other than zero, which might look like a bug. Like there are some historical versions I don't have access to.

Really good point -- I'll keep it at `0` and remove these `TODO`'s.


> While testing locally, I noticed duplicated rules (through bulk actions) have revision set to 1. So it seems like their revision gets immediately bumped after creation. When I create rules one by one they have revision set to 0.

Hmmm, were these rules being duplicated with additional reference objects like `actions` or `exceptions`? I came across this in some of the tests, but thought I had it working for the raw duplicate. Let me debug and see what I can do to keep it at `0` when duplicating regardless of if there are references or not.

Reply:
Okie, looks like it's not specific to the reference scenario, although that'll need to be fixed too (though only applicable for exceptions).

There's a call to `update()` after `create()` when duplicating that's supposed to handle the adding of the exceptions portion:

https://github.com/elastic/kibana/blob/8733774e8137fb531d4531fad58e09b65bba8689/x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/route.ts#L538-L547

However it looks like we're passing in a slightly different `riskScoreMapping` (`risk_score: undefined`) here which is resulting in the revision being incremented:

<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/2946766/220501649-6cc59ef7-4083-4724-ae1a-32295fbab4be.png" /> <img width="300" src="https://user-images.githubusercontent.com/2946766/220501688-47d79ece-ada0-4247-add6-38a819f5602b.png" />
</p>

I'll get a fix in place for this and be sure to add a test on the Security Solution side for this duplication logic. 👍

Reply:
Alrighty, fix is in place as of [f3b5c4e](https://github.com/elastic/kibana/pull/147398/commits/f3b5c4e580b6c5935ec3a42eaf513d0e2bcd38bd). I ended up adding a `shouldIncrementRevision` callback for consumers as we discussed in the other thread. I didn't fix this `risk_score_mapping` issue just yet as I want to check with @banderror to see why it [may've been typed this way](https://github.com/elastic/kibana/blob/21de750b3cff4d4ad27d07c77135327e644b3bee/packages/kbn-securitysolution-io-ts-alerting-types/src/risk_score_mapping/index.ts#L19). For now duplicates/clones will work as expected, but updates without changing anything will still result in an unexpected revision update until we resolve this.